### ISSUE TITLE
Installer is not built (gettext-cs branch)

### ISSUE THREAD

**Issue by adriangibanelbtactic** [serras/SparkleShare#2]
I think I have read the instructions ok (and improved them a bit while reading them ;) ) and here what does happen when you run:

```
sparkleshare\windows\build.cmd installer
```

```
Microsoft (R) Windows Installer Xml Compiler version 3.5.2519.0
Copyright (C) Microsoft Corporation. All rights reserved.

SparkleShare.wxs
candle.exe : error CNDL0103 : The system cannot find the file 'C:\serras\Sparkle
Share\SparkleShare\Windows\\msysgit.wxs' with type 'Source'.
candle.exe : error CNDL0103 : The system cannot find the file 'C:\serras\Sparkle
Share\SparkleShare\Windows\\po.wxs' with type 'Source'.
candle.exe : error CNDL0103 : The system cannot find the file 'C:\serras\Sparkle
Share\SparkleShare\Windows\\plugins.wxs' with type 'Source'.
light.exe : error LGHT0103 : The system cannot find the file 'msysgit.wixobj' wi
th type 'Source'.
SparkleShare.msi created.
```

So here there are two bugs:
- Installer is not buit.
- Script does  not actually if msi is created ok or not and says that the msi is created while it is not.

**Comment by serras** [comment]
There was an error in the previous instructions. You need to run the build command while being in the `sparkleshare\windows` folder (that was not that way before, I know). So the updated instructions read:

```
cd sparkleshare\windows
build installer
```

**Comment by serras** [comment]
About the MSI message, it's due to WiX, not to the actual script, so there's little we can do to not show the message :(

**Comment by adriangibanelbtactic** [comment]
Just to be sure...

Are you sure that the:

```
SparkleShare.msi created.
```

message is output from WiX and not from build script?

**Comment by serras** [comment]
I checked and indeed it is created by the build script. My fault :(
I'll try to add errorlevel checking after each of the scripts, but if run from the correct path, they should give no errors (they are just bundling whichever files they found in \bin directory).

**Comment by adriangibanelbtactic** [comment]
I have tried the updated Windows instructions. Windows 7 32 bit from nude install.

It seems there is a different error now.

```
Copyright (C) Microsoft Corporation. All rights reserved.

SparkleShare.wxs
Microsoft (R) Windows Installer Xml Compiler version 3.5.2519.0
Copyright (C) Microsoft Corporation. All rights reserved.

msysgit.wxs
Microsoft (R) Windows Installer Xml Compiler version 3.5.2519.0
Copyright (C) Microsoft Corporation. All rights reserved.

po.wxs
Microsoft (R) Windows Installer Xml Compiler version 3.5.2519.0
Copyright (C) Microsoft Corporation. All rights reserved.

plugins.wxs
Microsoft (R) Windows Installer Xml Linker version 3.5.2519.0
Copyright (C) Microsoft Corporation. All rights reserved.

C:\delivery\Dev\wix35_public\src\ext\UIExtension\wixlib\Common.wxs(15) : error L
GHT0103 : The system cannot find the file '..\..\data\install\bannrbmp.bmp'.
C:\delivery\Dev\wix35_public\src\ext\UIExtension\wixlib\Common.wxs(16) : error L
GHT0103 : The system cannot find the file '..\..\data\install\dlgbmp.bmp'.
SparkleShare.msi created.
```

**Comment by wimh** [comment]
bannrbmp.bmp and dlgbmp.bmp are missing too when I try. I think they are forgotten to add to git. A workaround is to delete the [two lines](https://github.com/serras/SparkleShare/blob/gettext-cs/SparkleShare/Windows/SparkleShare.wxs#L76) at the bottom of the SparkleShare.wxs file.

Meebey.SmartIrc4net.dll and Meebey.SmartIrc4net.xml are no longer required, so they can be removed

The wrong message `SparkleShare.msi created` can be fixed by deleting the file before the build:

```
if exist "%dp0\SparkleShare.msi" del "%dp0\SparkleShare.msi"
```

And checking if it exists afterwards:

```
if exist "%dp0\SparkleShare.msi" echo SparkleShare.msi created.
```

**Comment by Thirsty2** [comment]
I had similar problems mentioned above, and modified the build script to remove the two lines referencing bannrbmp.bmp and dlgbmp.bmp.  After that, it built.  However, when I run it, it asks me for my name and e-mail address and does not proceed when I hit the "Next" button.  I am stuck.  I tried running it as admin, but to no avail.  Next does nothing for me.

**Comment by wimh** [comment]
@Thirsty2  I just tried running the installer, and it seems to work for me. Exactly which version of Wix are you using?

**Comment by serras** [comment]
Thanks, I've added all your improvements to the repository, and also added the bitmap files I forgot previously.

@Thirsty2 Maybe you didn't add the msysgit folder correctly to the SparkleShare\bin folder and it cannot proceed because it cannot create your private and public keys.

**Comment by wimh** [comment]
@Thirsty2 forget my previous reply, I misunderstood your problem. But possibly next is disabled because the check whether the email address is valid fails. Try `a@b.cd`. Addresses like `first@last.name` or `me@playground.local` are not considered to be valid.

**Comment by Thirsty2** [comment]
I finally got back to this and tried a@b.cd, but still no luck.  FYI, my e-mail is an @gmail.com address, with only my username (only alpha chars) in front of the @.  It still hangs at the next button.  Perhaps I should note that I installed msysgit in its default location under program files months ago, and created my keys at that time, then recently copied it to the c:\msysgit folder after the fact without re-installing.  This was necessary to get it to build because of some hard coded paths in the script.  I noticed there have been quite a few check-ins since I tried getting this working, so I am tempted to pull the latest and rebuild, following the latest instructions.

**Comment by serras** [comment]
There has been a merged pull request just solving this problem. If you can, try with a newer version and report whether it works now.

**Comment by Thirsty2** [comment]
I got past the Next button, so the merged pull request seemed to work.  I was able to get sparkleshare installed and working with my gitolite server.  Very cool.  I successfully copied files into the sparkleshare dir, and they were easily pulled into another git repo after sparkleshare synced them to the server.  I like it.  Thanks for the help, and for all of your work on the Windows version of sparkleshare.
